src/lib/network_pool/snark_pool.ml

```diff
@@ -457,8 +457,19 @@ struct
                     false )
         in
         match One_or_two.zip proofs statements with
-        | Ok pairs ->
-            verify pairs
+        | Ok pairs -> (
+          match Signature_lib.Public_key.decompress prover with
```

Comment:
Do you think it would be better to add this check after other checks and before verifying? There could be many un-referenced works that can be eliminated without checking their key (assuming `decompress` would take longer than a hashtbl   membership check)